woocommerce/woocommerce
# [Product Block Editor] Add require password block field

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Create 'woocommerce/product-password-fields' block and use it on the block product editor



Closes #39127 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the block product editor.
2. Go to 'Organization' tab.
3. Check 'Require a password' checkbox.
4. See that the password field appears.
5. Fill in any value.
6. Save it and check that it is persisted. Optional: toggle to the legacy editor and see if the same value is there as well.
7. Uncheck the 'Require a password' checkbox.
8. Save it and see if the same state is persisted (The password field is cleared when it's hidden).



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
